README.md

```diff
@@ -279,6 +279,31 @@ role.
 Similar to `controller` and `vlan`, the `parent` references the connection profile in
 the ansible role.
 
+#### `type: infiniband`
+
+For the infiniband connection, the following options are supported:
+
+- `p_key`: The infiniband P_Key to use for the device. The default value of `-1` means
+  to use the P_Key at the index 0 (unset the p_key). Otherwise, it is a 16-bit unsigned
+  integer and the high bit should be set if it is a "full membership" P_Key.
+- `transport_mode`: The ipoib (IP OVER INFINIBAND) connection operation mode. The
+  possible modes are `datagram` and `connected`.
+
+**Note:** For the `p_key` setting, initscripts provider will normalize the `p_key`
+  internally to always set up the full membership bit (e.g. 0x000a will be normalized
+  into 0x800a) while nm provider does not. And both providers calculate the interface
+  name exactly the same way - based on the `parent` interface name and the P_Key, as a
+  result, `interface_name` will vary for two providers. (e.g. if the parent
+  `interface_name` is `ib0` and user defined `p_key` is 0x000a, the `interface_name`
+  for nm provider should be `ib0.000a`and the `interface_name` for initscripts provider
+  should be `ib0.800a`)
+
+**Note:** When using the initscripts provider, the script
```

Comment:
Just wondering, since the other code is fixing infiniband support. Does it mean that it never worked with the network role before? In that case, it seems feasible to only support it with the nm provider from now on.

Reply:
Correct,  the ipoib never worked before

Reply:
> Just wondering, since the other code is fixing infiniband support. Does it mean that it never worked with the network role before? In that case, it seems feasible to only support it with the nm provider from now on.

Did you mean to drop the infiniband tests for the initscripts provider?

Reply:
For configuring the network connection on the physical infiniband interface, it worked before. The ipoib (IP OVER INFINIBAND) is creating the connection on the virtual infiniband interface, it did not work before.

Reply:
Is the rdma package required for physical infiniband support, too?

Reply:
Yes, as long as `TYPE=Infiniband` specified in the ifcfg file, then ifup-ib (rdma package) is needed to support setting up the connection. Then I will drop the support for initscripts provider. Thanks.

Reply:
If phyiscal infiniband interfaces work with RHEL 7, then we cannnot drop support for that IMHO.

Reply:
But rdma-core package is not installed by default for initscripts provider in RHEL-7 though, it would not work for REHL 7 unless the users manually install the rdma-core package in the managed host themselves.